### PR TITLE
Feat/create user if not present

### PR DESCRIPTION
**Pull request by kglod-reef**

### CODE REVIEW

File: django_firebase_auth/views.py [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
-from firebase_admin import credentials, auth
-from firebase_admin.auth import ExpiredIdTokenError
-from firebase_admin.exceptions import FirebaseError
+from django.contrib.auth.models import AbstractBaseUser
 
-if settings.GOOGLE_SERVICE_ACCOUNT_FILE:
-    firebase_credentials = credentials.Certificate(settings.GOOGLE_SERVICE_ACCOUNT_FILE)
-    firebase_admin.initialize_app(firebase_credentials)
+AUTH_BACKEND = settings.DJANGO_FIREBASE_AUTH_AUTH_BACKEND
+SERVICE_ACCOUNT_FILE = settings.DJANGO_FIREBASE_AUTH_SERVICE_ACCOUNT_FILE
```
Comment (kglod-reef) [inline django_firebase_auth/views.py:14]: those two settings are required so when they are not present we should fail server initialization

File: django_firebase_auth/views.py [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+JWT_HEADER_NAME = getattr(settings, "DJANGO_FIREBASE_AUTH_JWT_HEADER_NAME", "X-FIREBASE-JWT")
+CREATE_USER_IF_NOT_EXISTS = getattr(settings, "DJANGO_FIREBASE_AUTH_CREATE_USER_IF_NOT_EXISTS", False)
+ALLOW_NOT_CONFIRMED_EMAILS = getattr(settings, "DJANGO_FIREBASE_AUTH_ALLOW_NOT_CONFIRMED_EMAILS", False)
```
Comment (kglod-reef) [inline django_firebase_auth/views.py:17]: I renamed all of those settings to have the same `DJANGO_FIREBASE_AUTH_` prefix. It's pretty long but at least there is no risk of clash

File: django_firebase_auth/views.py [hunk 3]
```diff
@@ -45,8 +48,36 @@ class EmailNotVerified(AuthError):
...
+    try:
+        return UserModel.objects.get(email=email)
+    except UserModel.DoesNotExist:
+        if not CREATE_USER_IF_NOT_EXISTS:
+            return None
+
+    user = UserModel.objects.create_user(email=email, is_active=True)
+    user.set_unusable_password()
+    user.save()
+    return user
```
Comment (kglod-reef) [inline django_firebase_auth/views.py:76]: default User model does not provide any `email_verified` flag so I'm not handling it here

File: django_firebase_auth/views.py [hunk 4]
```diff
@@ -45,8 +48,36 @@ class EmailNotVerified(AuthError):
...
+        email = _verify_firebase_account(request.headers)
+    except AuthError as ex:
+        return JsonResponse(ex.make_response_body(), status=401)
+
+    user = _get_or_create_user(email)
+    if user is None:
+        return JsonResponse(UserNotRegistered.make_response_body(), status=401)
+
+    login(request=request, user=user, backend=AUTH_BACKEND)
+    return JsonResponse({"status": "ok"})
```
Comment (kglod-reef) [inline django_firebase_auth/views.py:62]: I changed result to JSON to be compatible with error responses

**Review by mpnowacki-reef — APPROVED** [review]

**Pull request merged**